applications/GeoMechanicsApplication/custom_processes/set_parameter_field_process.cpp

```diff
@@ -0,0 +1,155 @@
+//                   Kratos default license: kratos/license.txt
+//
+//  Main authors:    Aron Noordam
+//
+
+// System includes
+
+// External includes
+
+// Project includes
+#include "set_parameter_field_process.hpp"
+
+#include <utilities/function_parser_utility.h>
+#include <utilities/mortar_utilities.h>
+
+#include "utilities/interval_utility.h"
+#include "geo_mechanics_application_variables.h"
+
+
+namespace Kratos
+{
+
+    SetParameterFieldProcess::SetParameterFieldProcess(ModelPart& rModelPart,
+                                                       const Parameters Settings)
+                                                            : Process(), mrModelPart(rModelPart),
+                                                            mParameters(Settings)
+{
+    // function type: python, cpp, input
+    const Parameters default_parameters(R"(
+        {
+            "help"            : "This process applies a moving load condition belonging to a modelpart. The load moves over line elements.",
+            "model_part_name" : "please_specify_model_part_name",
+            "variable_name"   : "CUSTOM",
+            "func_type"       : "input",               
+            "function"        : "0",
+            "dataset"         : "dummy"
+        }  )"
+    );
+
+    mParameters.RecursivelyValidateAndAssignDefaults(default_parameters);
+
+}
+
+
+void SetParameterFieldProcess::SetValueAtElement(Element& rElement, const Variable<double>& rVar, const double Value)
+{
+
+    Properties& r_prop = rElement.GetProperties();
+
+    // Copies properties
+    Properties::Pointer p_new_prop = Kratos::make_shared<Properties>(r_prop);
+
+    // Adds new properties to the element
+    p_new_prop->SetValue(rVar, Value);
+    rElement.SetProperties(p_new_prop);
+
+}
+
+
+void SetParameterFieldProcess::SetParameterFieldUsingInputFunction(const Variable<double>& rVar)
+{
+    auto parameter_function = BasicGenericFunctionUtility(mParameters["function"].GetString());
+    const double current_time = this->mrModelPart.GetProcessInfo().GetValue(TIME);
+
+    for (Element& r_element : mrModelPart.Elements()) {
+
+        const auto& r_geom = r_element.GetGeometry();
+
+        // calculate parameter value at current element
+        const double val = parameter_function.CallFunction(r_geom.Center().X(), r_geom.Center().Y(), r_geom.Center().Z(), current_time, 0, 0, 0);
+
+        SetValueAtElement(r_element, rVar, val);
+
+    }
+}
+
+
+void SetParameterFieldProcess::SetParameterFieldUsingPythonFunction(const Variable<double>& rVar)
+{
+    // get new data from the data set
+    const std::string dataset = mParameters["dataset"].GetString();
+    const Parameters new_data{ dataset };
+    const Vector data_vector = new_data["values"].GetVector();
+
+    // set new data on the elements
+    IndexType i = 0;
+    for (Element& r_element : mrModelPart.Elements())
+    {
+        SetValueAtElement(r_element, rVar, data_vector[i]);
+        i++;
```

Comment:
You can compact these  two lines:
```suggestion
        SetValueAtElement(r_element, rVar, data_vector[i++]);
```